### PR TITLE
feat(ctb): Consolidate lock files into a single directory

### PR DESCRIPTION
**Pull request by maurelian**
**Description**

Does the following in the contracts package: 

1. moves all lock/snapshot files to a single `locks` dir. This will ensure that these files appear together during PR reviews. 
2. uses the term 'lock' more consistently as a means of codifying this pattern.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -382,10 +382,10 @@ jobs:
             pnpm lint:check || echo "export LINT_STATUS=1" >> "$BASH_ENV"
           working_directory: packages/contracts-bedrock
       - run:
-          name: gas snapshot
+          name: gas lock
           command: |
             forge --version
-            pnpm gas-snapshot --check || echo "export GAS_SNAPSHOT_STATUS=1" >> "$BASH_ENV"
+            pnpm gas-lock --check || echo "export GAS_LOCK_STATUS=1" >> "$BASH_ENV"
```
Comment (maurelian) [inline .circleci/config.yml:388]: this kind of breaks the standard `gas-snapshot` terminology, so I'd be OK with reverting this.

Comment (clabby) [reply .circleci/config.yml:388]: Not opinionated here tbh. Both mean the same thing in my head 😄

File: packages/contracts-bedrock/scripts/storage-lock.sh [hunk 2]
```diff
@@ -38,15 +38,15 @@ dir=$(dirname "$0")
 
 echo "Creating storage layout diagrams.."
 
-echo "=======================" > $dir/../.storage-layout
-echo "👁👁 STORAGE LAYOUT snapshot 👁👁" >> $dir/../.storage-layout
-echo "=======================" >> $dir/../.storage-layout
+echo "=======================" > $dir/../.storage-lock
+echo "👁👁 STORAGE LAYOUT LOCK 👁👁" >> $dir/../.storage-lock
+echo "=======================" >> $dir/../.storage-lock
```
Comment (clabby) [inline packages/contracts-bedrock/scripts/storage-lock.sh]:
```suggestion
echo "=======================" > $dir/../locks/storage-lock
echo "👁👁 STORAGE LAYOUT LOCK 👁👁" >> $dir/../locks/storage-lock
echo "=======================" >> $dir/../locks/storage-lock
```

File: packages/contracts-bedrock/scripts/storage-lock.sh [hunk 3]
```diff
@@ -38,15 +38,15 @@ dir=$(dirname "$0")
...
 for contract in ${contracts[@]}
 do
-  echo -e "\n=======================" >> $dir/../.storage-layout
-  echo "➡ $contract">> $dir/../.storage-layout
-  echo -e "=======================\n" >> $dir/../.storage-layout
-  forge inspect --pretty $contract storage-layout >> $dir/../.storage-layout
+  echo -e "\n=======================" >> $dir/../.storage-lock
+  echo "➡ $contract">> $dir/../.storage-lock
+  echo -e "=======================\n" >> $dir/../.storage-lock
+  forge inspect --pretty $contract storageLayout >> $dir/../.storage-lock
```
Comment (clabby) [inline packages/contracts-bedrock/scripts/storage-lock.sh]:
```suggestion
  echo -e "\n=======================" >> $dir/../locks/storage-lock
  echo "➡ $contract">> $dir/../locks/storage-lock
  echo -e "=======================\n" >> $dir/../locks/storage-lock
  forge inspect --pretty $contract storageLayout >> $dir/../locks/storage-lock
```

File: packages/contracts-bedrock/scripts/storage-lock.sh [hunk 4]
```diff
@@ -38,15 +38,15 @@ dir=$(dirname "$0")
...
 done
-echo "Storage layout snapshot stored at $dir/../.storage-layout"
+echo "Storage layout lock stored at $dir/../.storage-lock"
```
Comment (clabby) [inline packages/contracts-bedrock/scripts/storage-lock.sh]:
```suggestion
echo "Storage layout lock stored at $dir/../locks/storage-lock"
```

**Review by clabby — CHANGES_REQUESTED** [review]
Looks great, nice cleanup. Needs a couple edits to `storage-lock.sh`, will approve after :)

**Comment by maurelian** [comment]
Good catch on those broken paths, thanks @clabby.

**Comment by clabby** [comment]
Can't leave an inline comment because it's out of scope of the changes, but in the circleci config, the result of the gas snapshot check is currently ignored. The result is now stored in an env var called `GAS_LOCK_STATUS`, but the conditional in the `contracts-bedrock-checks` job checks for the `GAS_SNAPSHOT_STATUS` (which doesn't exist anymore)

(L425 of the circleci config)

**Review by clabby — APPROVED** [review]
LGTM!

**Review by twoshark — APPROVED** [review]

**Review by refcell — APPROVED** [review]
This is awesome

**Comment by refcell** [comment]
Merged my small fix since CI passed 😄

**Pull request merged**